src/routes/events/containers/__tests__/EventsPage.spec.js

```diff
@@ -0,0 +1,255 @@
+    )
+  })
+
+  test('should not dispatch on prop update', () => {
+    const mockSetLanguageChangeUrls = jest.fn()
+
+    const wrapper = shallow(
+      <EventsPage events={events}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  id={id}
+                  setLanguageChangeUrls={mockSetLanguageChangeUrls} />
+    )
+
+    let mockCalls = mockSetLanguageChangeUrls.mock.calls
+
+    wrapper.setProps({events: events, ...wrapper.props})
+
+    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(mockCalls.length)
+
+    wrapper.setProps({...wrapper.props})
+
+    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(mockCalls.length)
+  })
+
+  test('mapLanguageToUrl', () => {
+    const mockSetLanguageChangeUrls = jest.fn()
+
+    const mapLanguageToUrl = shallow(
+      <EventsPage events={events}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  id={id}
+                  setLanguageChangeUrls={mockSetLanguageChangeUrls} />
+    ).instance().mapLanguageToUrl
+
+    expect(mapLanguageToUrl('en')).toBe('/augsburg/en/events')
+    expect(mapLanguageToUrl('en', 1234)).toBe('/augsburg/en/events/1234')
+  })
+
+  const mockStore = configureMockStore([thunk])
+
+  describe('connect', () => {
+    test('should map state to props', () => {
+      const store = mockStore({
+        events: new Payload(false),
+        languages: new Payload(false),
+        router: {params: {location: 'augsburg', language: 'en', id: '1234'}}
+      })
+
+      const tree = mount(
+        <Provider store={store}>
+          <ConnectedEventsPage />
+        </Provider>
+      )
+
+      const eventsPageProps = tree.find(ConnectedEventsPage).childAt(0).props()
+
+      // todo add events and languages
+      expect(eventsPageProps).toEqual({
+        location: 'augsburg',
+        language: 'en',
+        id: '1234',
+        setLanguageChangeUrls: expect.any(Function)
+      })
+    })
+
+    test('should map dispatch to props', () => {
+      const store = mockStore({
+        events: new Payload(false),
+        languages: new Payload(false),
+        router: {params: {location: location, language: language}}
+      })
+
+      const mapLanguageToUrl = (language, id) => 'test' + language + id
+
+      const testUrls = {
+        en: 'testenundefined',
+        de: 'testde1235',
+        ar: 'testar1236'
+      }
+
+      const availableLanguages = {
+        de: '1235',
+        ar: '1236'
+      }
+
+      const tree = mount(
+        <Provider store={store}>
+          <ConnectedEventsPage />
+        </Provider>
+      )
+
+      // todo expect setLanguageChangeUrls action to be in store, but as we don't get events and languages from our
+      // mocked endpoint no action is dispatched
+
+      const eventsPageProps = tree.find(ConnectedEventsPage).childAt(0).props()
```

Comment:
We should finish the testing for the passing of the props in this task. The next tasks will need a working solution.

Did you already try the solution with the promises and real store?

Reply:
yeah I tried but couldn't get it to work

Reply:
created a task